contracts/volt/VoltV2.sol

```diff
@@ -0,0 +1,549 @@
+            return 0;
+        }
+
+        // First check most recent balance
+        if (checkpoints[account][nCheckpoints - 1].fromBlock <= blockNumber) {
+            return checkpoints[account][nCheckpoints - 1].votes;
+        }
+
+        // Next check implicit zero balance
+        if (checkpoints[account][0].fromBlock > blockNumber) {
+            return 0;
+        }
+
+        uint32 lower = 0;
+        uint32 upper = nCheckpoints - 1;
+        while (upper > lower) {
+            uint32 center = upper - (upper - lower) / 2; // ceil, avoiding overflow
+            Checkpoint memory cp = checkpoints[account][center];
+            if (cp.fromBlock == blockNumber) {
+                return cp.votes;
+            } else if (cp.fromBlock < blockNumber) {
+                lower = center;
+            } else {
+                upper = center - 1;
+            }
+        }
+        return checkpoints[account][lower].votes;
+    }
+
+    function _burn(address src, uint96 amount) internal {
+        require(src != address(0), "Volt: cannot burn from the zero address");
+        require(balances[src] >= amount, "Volt: burn amount exceeds balance");
+
+        uint96 safeSupply = safe96(
+            totalSupply,
+            "Volt: totalSupply exceeds 96 bits"
+        );
+
+        totalSupply = sub96(safeSupply, amount, "Volt: subtraction underflow");
+
+        balances[src] = sub96(
+            balances[src],
+            amount,
+            "Volt: subtraction underflow"
+        );
+
+        emit Transfer(src, address(0), amount);
+
+        _moveDelegates(delegates[src], address(0), amount);
+    }
+
+    function _spendAllowance(address src, uint256 rawAmount) internal {
+        address spender = msg.sender;
+        uint96 spenderAllowance = allowances[src][spender];
+        uint96 amount = safe96(rawAmount, "Volt: amount exceeds 96 bits");
+
+        if (spender != src && spenderAllowance != type(uint96).max) {
+            uint96 newAllowance = sub96(
+                spenderAllowance,
+                amount,
+                "Volt: transfer amount exceeds spender allowance"
+            );
+            allowances[src][spender] = newAllowance;
+
+            emit Approval(src, spender, newAllowance);
+        }
+    }
+
+    function _delegate(address delegator, address delegatee) internal {
+        address currentDelegate = delegates[delegator];
+        uint96 delegatorBalance = balances[delegator];
+        delegates[delegator] = delegatee;
+
+        emit DelegateChanged(delegator, currentDelegate, delegatee);
+
+        _moveDelegates(currentDelegate, delegatee, delegatorBalance);
+    }
+
+    function _transferTokens(
+        address src,
+        address dst,
+        uint96 amount
+    ) internal {
+        require(
+            src != address(0),
+            "Volt: cannot transfer from the zero address"
+        );
+        require(dst != address(0), "Volt: cannot transfer to the zero address");
```

Comment:
We can probably remove this check as it eats up gas and doesn't prevent you from setting address(1) or any similarly invalid address.